lib/publiccloud/instance.pm

```diff
@@ -349,6 +349,36 @@ sub wait_for_guestregister {
     die('guestregister didn\'t end in expected timeout=' . $args{timeout});
 }
 
+=head2 update_instance_ip
+
+    update_instance_ip(timeout => 600)
+
+This subroutine checks the public IP cloud provider provides for the VM.
+When the IP differs from `$self->public_ip` we update `$self->public_ip`.
+
+=cut
+
+sub update_instance_ip {
+    my $self = shift;
+    my $timeout = 300;
+    my $delay = 5;
+
+    return if (get_var('PUBLIC_CLOUD_SLES4SAP'));
```

Comment:
Final nitpick, sorry:
what if moving this if-return on top and compact all `my ...`: if SAP no assignment? Up to you.

Reply:
That's against the code style we follow.
Statically variables are always assigned on top.

Sorry.